tests/components/sensor/test_arlo.py

```diff
@@ -0,0 +1,185 @@
+from unittest.mock import patch, MagicMock
+from homeassistant.const import (
+    DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_HUMIDITY, ATTR_ATTRIBUTION)
+from homeassistant.components.sensor import arlo
+from homeassistant.components.arlo import DATA_ARLO
+from homeassistant.helpers import dispatcher
+from tests.common import get_test_home_assistant
+
+
+class TestArloSensor(unittest.TestCase):
+    """Test Netgear Arlo sensors."""
+
+    def _setup_arlo(self):
+        self.hass.data = {
+            DATA_ARLO: 'test'
+        }
+
+    @staticmethod
+    def _get_mock_sensor(name='Last', sensor_type='last_capture', data=None):
+        if data is None:
+            data = {}
+        return arlo.ArloSensor(name, data, sensor_type)
+
+    @staticmethod
+    def _get_named_tuple(input_dict):
+        return namedtuple('Struct', input_dict.keys())(*input_dict.values())
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup shared dependencies for each test."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Tear down shared dependencies for each test."""
+        self.hass.stop()
+
+    def test_setup_with_no_data(self):
+        """Test setup_platform with no data."""
+        result = arlo.setup_platform(self.hass, None, None)
+        self.assertEqual(result, False)
+
+    sensors = None
+
+    def _add_devices(self, sensors, boolean):
+        if boolean:
+            self.sensors = sensors
+
+    def test_setup_with_valid_data(self):
+        """Test setup_platform with valid data."""
+        config = {
+            'monitored_conditions': [
+                'last_capture',
+                'total_cameras',
+                'captured_today',
+                'battery_level',
+                'signal_strength',
+                'temperature',
+                'humidity',
+                'air_quality'
+            ]
+        }
+
+        self.hass.data[DATA_ARLO] = self._get_named_tuple({
+            'cameras': [self._get_named_tuple({
+                'name': 'Camera',
+                'model_id': 'ABC1000'
+            })],
+            'base_stations': [self._get_named_tuple({
+                'name': 'Base Station',
+                'model_id': 'ABC1000'
+            })]
+        })
+
+        arlo.setup_platform(self.hass, config, self._add_devices)
+        self.assertEqual(len(self.sensors), 8)
+
+    def test_sensor_name(self):
+        """Test the name property."""
+        sensor = TestArloSensor._get_mock_sensor()
+        self.assertEqual(sensor.name, 'Last')
+
+    @asyncio.coroutine
+    @patch('homeassistant.helpers.dispatcher.async_dispatcher_connect',
+           MagicMock())
+    async def test_async_added_to_hass(self):
+        """Test dispatcher called when added."""
+        sensor = TestArloSensor._get_mock_sensor()
+        await sensor.async_added_to_hass()
+        dispatcher.async_dispatcher_connect.assert_called_once()
```

Comment:
We prefer not to use `assert_called_once` but instead use `assert len(mock.calls) == 1`. A typo on the mock method would pass silently.